File: content/problems/1185-day-of-the-week/solutions/days-since-known-friday/Solution.java
class Solution {
    public String dayOfTheWeek(int day,int month,int year){
        String[] names={
            "Sunday","Monday","Tuesday","Wednesday","Thursday","Friday","Saturday"
        };
        int days=0;
        for(int current=1971;current<year;current++)days+=leap(current)?366:365;
        int[] monthDays={
            31,28,31,30,31,30,31,31,30,31,30,31
        };
        for(int current=1;current<month;current++)days+=monthDays[current-1]+(current==2&&leap(year)?1:0);
        days+=day-1;
        return names[(5+days)%7];
    }
    private boolean leap(int year){
        return year%400==0||year%4==0&&year%100!=0;
    }
}

File: content/problems/1185-day-of-the-week/solutions/days-since-known-friday/Solution.kt
class Solution {
    fun dayOfTheWeek(day:Int,month:Int,year:Int):String{
        val names=arrayOf("Sunday","Monday","Tuesday","Wednesday","Thursday","Friday","Saturday");
        var days=0;
        for(current in 1971 until year)days+=if(leap(current))366 else 365;
        val monthDays=intArrayOf(31,28,31,30,31,30,31,31,30,31,30,31);
        for(current in 1 until month)days+=monthDays[current-1]+if(current==2&&leap(year))1 else 0;
        days+=day-1;
        return names[(5+days)%7]
    };
    private fun leap(year:Int)=year%400==0||year%4==0&&year%100!=0
}
